cli/src/config.rs

```diff
@@ -127,28 +130,7 @@ impl ProjectConfig {
     }
 }
 
-/// Create or open a file. If the file is created, it will restrict permissions
-/// to allow read/write access only to the current user.
-fn create_private_file<P: AsRef<Path>>(path: P) -> io::Result<fs::File> {
-    // Use OpenOptions so that we can specify the permission bits
-    let mut opts = fs::OpenOptions::new();
-    opts.write(true).create(true).truncate(true);
-    #[cfg(unix)]
-    {
-        opts.mode(0o600);
-    }
-
-    // Windows file permissions are complicated and home folders aren't usually
-    // globally readable, so we can ignore Windows for now.
```

Comment:
The first half of this comment appears to be the reason the config tests are failing for Windows. It looks like ignoring it will no longer be possible.

Reply:
I'm not entirely sure what this comment means by "globally readable", but I don't think that should be cause for this issue.

Reply:
I suspect the Windows tests are failing because  something (probably another test running in parallel) still has an open handle on the old config file when the `fs::rename()` happens.